platform/ios/src/MGLMapView.mm

```diff
@@ -374,10 +375,23 @@ - (void)commonInit
     _compassView.alpha = 0;
     _compassView.userInteractionEnabled = YES;
     [_compassView addGestureRecognizer:[[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(handleCompassTapGesture:)]];
-    UIView *container = [[UIView alloc] initWithFrame:CGRectZero];
-    [container addSubview:_compassView];
-    container.translatesAutoresizingMaskIntoConstraints = NO;
-    [self addSubview:container];
+    _compassView.translatesAutoresizingMaskIntoConstraints = NO;
+    [NSLayoutConstraint constraintWithItem:_compassView
+                              attribute:NSLayoutAttributeWidth
+                              relatedBy:NSLayoutRelationEqual
+                                 toItem:nil
+                              attribute:NSLayoutAttributeNotAnAttribute
+                             multiplier:1
+                               constant:_compassView.image.size.width];
+    [NSLayoutConstraint constraintWithItem:_compassView
+                              attribute:NSLayoutAttributeHeight
+                              relatedBy:NSLayoutRelationEqual
+                                 toItem:nil
+                              attribute:NSLayoutAttributeNotAnAttribute
+                             multiplier:1
+                               constant:_compassView.image.size.height];
```

Comment:
These constraints are never added to a view so ARC will have them removed after this line.
They're  also unnecessary since the UIImageView is initialized with `-initWithImage` which handles the size in `-intrinsicContentSize` IIRC. This also means that line: 374 `_compassView.frame = CGRectMake(` is redundant and can be removed.